commit bd9b4e60690281a2e2f555d7d6e7dc9afcaa3f44
Author: Alagu Sankar <alagusankar@silex-india.com>
Date:   Fri Jun 29 16:27:56 2018 +0300

    ath10k: sdio: use same endpoint id for all packets in a bundle
    
    [ Upstream commit 679e1f07c86221b7183dd69df7068fd42d0041f6 ]
    
    All packets in a bundle should use the same endpoint id as the
    first lookahead.
    
    This matches how things are done is ath6kl, however,
    this patch can theoretically handle several bundles
    in ath10k_sdio_mbox_rx_process_packets().
    
    Without this patch we get lots of errors about invalid endpoint id:
    
    ath10k_sdio mmc2:0001:1: invalid endpoint in look-ahead: 224
    ath10k_sdio mmc2:0001:1: failed to get pending recv messages: -12
    ath10k_sdio mmc2:0001:1: failed to process pending SDIO interrupts: -12
    
    Co-Developed-by: Niklas Cassel <niklas.cassel@linaro.org>
    Signed-off-by: Alagu Sankar <alagusankar@silex-india.com>
    Signed-off-by: Niklas Cassel <niklas.cassel@linaro.org>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/wireless/ath/ath10k/sdio.c b/drivers/net/wireless/ath/ath10k/sdio.c
index d612ce8c9cff..d46523b0472c 100644
--- a/drivers/net/wireless/ath/ath10k/sdio.c
+++ b/drivers/net/wireless/ath/ath10k/sdio.c
@@ -434,12 +434,14 @@ static int ath10k_sdio_mbox_rx_process_packets(struct ath10k *ar,
 	enum ath10k_htc_ep_id id;
 	int ret, i, *n_lookahead_local;
 	u32 *lookaheads_local;
+	int lookahead_idx = 0;
 
 	for (i = 0; i < ar_sdio->n_rx_pkts; i++) {
 		lookaheads_local = lookaheads;
 		n_lookahead_local = n_lookahead;
 
-		id = ((struct ath10k_htc_hdr *)&lookaheads[i])->eid;
+		id = ((struct ath10k_htc_hdr *)
+		      &lookaheads[lookahead_idx++])->eid;
 
 		if (id >= ATH10K_HTC_EP_COUNT) {
 			ath10k_warn(ar, "invalid endpoint in look-ahead: %d\n",
@@ -462,6 +464,7 @@ static int ath10k_sdio_mbox_rx_process_packets(struct ath10k *ar,
 			/* Only read lookahead's from RX trailers
 			 * for the last packet in a bundle.
 			 */
+			lookahead_idx--;
 			lookaheads_local = NULL;
 			n_lookahead_local = NULL;
 		}
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1139_bd9b4e60690281a2e2f555d7d6e7dc9afcaa3f44.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1139_bd9b4e60690281a2e2f555d7d6e7dc9afcaa3f44.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
